src/content/monitoring/_index.en.md

```diff
@@ -0,0 +1,57 @@
++++
+title = "Monitoring"
+date = 2020-08-12T16:02:00+02:00
+weight = 20
+pre = "<b>4. </b>"
++++
+
+## Basic Overview
+
+Submariner provides a number of Prometheus metrics, and sets up `ServiceMonitor` instances which allow these metrics to be scraped by an
+in-cluster Prometheus instance.
+
+The following metrics are exposed currently:
+
+* `submariner_gateways`, the number of gateways in the cluster;
+
+* `submariner_connections`, the number of connections to other clusters, with the following labels:
+
+  * `local_cluster`, the local cluster name
+  * `local_hostname`, the local hostname
+  * `remote_cluster`, the remote cluster name
+  * `remote_hostname`, the remote hostname
+  * `status` the connection status (“connecting”, “connected”, or “error”)
+
+### OpenShift setup
+
+OpenShift 4.5 or later can automatically discover the Submariner metrics.
+This currently requires enabling user workload monitoring; see
+[the OpenShift documentation](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.5/html/monitoring/monitoring-your-own-services)
+for details.
+
+### Prometheus operator
+
+To start monitoring Submariner using the Prometheus operator, Prometheus needs to be configured to scrape the operator’s namespace
+(`submariner-operator` by default). The specifics depend on your Prometheus deployment, but typically, this will involve:
+
+* adding the operator’s namespace to Prometheus’ `ClusterRoleBinding`;
+
+* ensuring that Prometheus’ configuration doesn’t prevent it from scraping this namespace.
+
+A minimal `Prometheus` object providing access to the Submariner metrics is as follows:
```

Comment:
I assume the user must create the resource.

```suggestion
A `Prometheus` resource must be created to provide access to the Submariner metrics, for example:
```

Reply:
I wonder if we could automate this from the operator too :)

Reply:
I’ve rephrased the introductory paragraph to make it clear that the administrator needs to take care of both.

@mangelajo I considered that, but there are many different Prometheus deployments; we could try to figure out if the target is the default OpenShift setup, and configure that, but I don’t know what OpenShift “thinks” of tools changing its `ClusterRoleBinding`s behind its back ;-).

Reply:
Duh, I forgot I documented OpenShift in the previous paragraph. The main issue here is that the `Prometheus` object specifies the overall Prometheus deployment; I think we should leave that up to the administrator...

Reply:
I think we should eventually automate that, it makes sense, don't see why it should be a problem. We don't need to change it's ClusterRoleBinding, we can add the permissions to our own cluster roles

Reply:
Automating `Prometheus` setup is a bit like signing up with an ISP, and suddenly having workers show up to install an alarm and an extension to your home. I’m not sure how changing our own cluster roles helps, that wouldn’t give any extra permissions to Prometheus... (Prometheus pulls data; we don’t send data to Prometheus, it retrieves data from us.)

Reply:
> Automating `Prometheus` setup is a bit like signing up with an ISP, and suddenly having workers show up to install an alarm and an extension to your home. I’m not sure how changing our own cluster roles helps, that wouldn’t give any extra permissions to Prometheus... (Prometheus pulls data; we don’t send data to Prometheus, it retrieves data from us.)

We could have a flag on ... if you want that done for you, or not... :)